api/ocm/tools/transfer/transfer.go

```diff
@@ -221,17 +220,20 @@ func transferVersion(ctx context.Context, log logging.Logger, state WalkingState
 }
 
 func CopyVersion(printer common.Printer, log logging.Logger, hist common.History, src ocm.ComponentVersionAccess, t ocm.ComponentVersionAccess, handler TransferHandler) (rerr error) {
-	return copyVersion(context.Background(), common.AssurePrinter(printer), log, hist, src, t, src.GetDescriptor().Copy(), handler)
+	return CopyVersionWithContext(context.Background(), printer, log, hist, src, t, handler)
 }
 
-func CopyVersionWithContext(cctx context.Context, log logging.Logger, hist common.History, src ocm.ComponentVersionAccess, t ocm.ComponentVersionAccess, handler TransferHandler) (rerr error) {
-	return copyVersion(cctx, common.GetPrinter(cctx), log, hist, src, t, src.GetDescriptor().Copy(), handler)
+func CopyVersionWithContext(cctx context.Context, printer common.Printer, log logging.Logger, hist common.History, src ocm.ComponentVersionAccess, t ocm.ComponentVersionAccess, handler TransferHandler) (rerr error) {
+	return copyVersionWithWorkerPool(cctx, printer, log, hist, src, t, src.GetDescriptor().Copy(), handler, 5)
```

Comment:
The 5 workers seem very specific to your trial run and should not be used for OCM in general. 

I have 2 suggestions for possible value setting:

1. Default to a runtime-dynamic value, for example `runtime.NumCPU()` that defaults to active discovered CPUs by the GO runtime
2. Default to a sequential value, i.e. 1

Choose whichever you prefer.

That being said, to properly make it configurable like you describe in your PR, I am missing exactly that: a configurable flag. As such you will need to introduce both a setting/option as well as a CLI flag for concurrency. Ideally this should also be configurable in .ocmconfig. I don't mind for now if you want to have this in separate PRs but imo the merge should not happen without ANY configuration ability.

Thanks!